### ISSUE TITLE
PRIMITIVE(grant_access) leaks memory when it throws

### ISSUE THREAD

**Issue by erikcorry** [toitlang/toit#1975]
```
PRIMITIVE(grant_access) {
  PRIVILEGED;
  ARGS(int, client, int, handle, uword, offset, uword, size, bool, writable);
  RegionGrant* grant = _new RegionGrant(client, handle, offset, size, writable);
  if (!grant) FAIL(MALLOC_FAILED);
  Locker locker(OS::global_mutex());
  for (auto it : grants) {
    if (it->offset() == offset && it->size() == size) FAIL(ALREADY_IN_USE);  // !!!!!!! Leaks grant.
  }
  grants.prepend(grant);
  return process->null_object();
}
```

**Comment by kasperl** [comment]
It would have taken you less time to fix it than to file the issue ;)

**Comment by kasperl** [comment]
Fixed in #1976.
